### PR TITLE
{tools}[dummy,intel/2015b] hanythingondemand 3.0.0 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
still WIP because:
* final hanythingondemand 3.0.0 is not available yet (still in development)
* these easyconfigs contain site-specific stuff (see `modextravars` in the 'minimal' `hanythingondemand` easyconfig)

cc @ehiggs

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: test this pleas

**Comment by boegel** [comment]
Jenkins: test this please

File: easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-minimal.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+use_easy_install = True
+zipped_egg = True
+
+modaltsoftname = 'hod'
+
+options = {'modulename': 'hod'}
+
+modextravars = {
+    'HOD_BATCH_WORKDIR': '\$VSC_SCRATCH/hod',
```
Comment (boegel) [inline easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-minimal.eb]: site-specific, this should not be included in here like this

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 9 out of 9 (9 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b9ffefea7313aaf1e622 for a full test report.

File: easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-cli.eb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+use_easy_install = True
+zipped_egg = True
+
+# use 'hod' rather than 'hanythingondemand' in module name
+modaltsoftname = 'hod'
+
+options = {'modulename': 'hod'}
+
+modextravars = {
+    'HOD_BATCH_HOD_MODULE': 'hanythingondemand/3.0.0-beta-intel-2015b-Python-2.7.10',
```
Comment (wpoely86) [inline easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-cli.eb]: hardcoded?

**Comment by wpoely86** [comment]
looks good

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/bce356a55b140d75117f for a full test report.

File: easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-cli.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-cli.eb]: I could use `%(version)s` here indeed, you're right, thanks!

Comment (boegel) [inline easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-cli.eb]: ah, and yes, hmm, the module name is indeed hardcoded, but others may be using a different naming scheme here, point taken...

I'll comment this out then, we can modify as needed when actually installing it (and HOD will work without having these set, as long as you use `--workdir` and `--hod-module`, and if not, it'll tell you).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/c77992ea3691cdd6e8bd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2331bbe9d9d5c3671c22 for a full test report.

File: easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-intel-2015b-Python-2.7.10.eb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+# a Python version with netaddr/netifaces/mpi4py + pbs_python and vsc-base/vsc-mympirun is required at runtime
+dependencies = [
+    (python, pythonver),  # provides netaddr, netifaces, mpi4py
+    ('pbs_python', '4.6.0', versionsuffix),
+    ('vsc-mympirun', '3.4.2', versionsuffix + '-vsc-base-2.4.2'),
+    ('setuptools', '1.4.2', '', True),
+]
+
+use_easy_install = True
```
Comment (ehiggs) [inline easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-intel-2015b-Python-2.7.10.eb]: if the setup.py has `zip_safe=True` this shouldn't be needed.

File: easybuild/easyconfigs/v/vsc-base/vsc-base-2.4.2-intel-2015b-Python-2.7.10.eb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+easyblock = 'PythonPackage'
+
+name = 'vsc-base'
+version = '2.4.2'
```
Comment (ehiggs) [inline easybuild/easyconfigs/v/vsc-base/vsc-base-2.4.2-intel-2015b-Python-2.7.10.eb]: We don't use this. We use 2.2.2.

File: easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-intel-2015b-Python-2.7.10.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-intel-2015b-Python-2.7.10.eb]: shouldn't, but I'm not 100% sure; at least this makes very sure we're using `easy_install`, which is important for `hod dists`, see also https://github.com/hpcugent/hanythingondemand/issues/78

File: easybuild/easyconfigs/v/vsc-base/vsc-base-2.4.2-intel-2015b-Python-2.7.10.eb (hunk 4 above)
Comment (boegel) [inline easybuild/easyconfigs/v/vsc-base/vsc-base-2.4.2-intel-2015b-Python-2.7.10.eb]: We *do* use 2.4.2 in both hanythingondemand easyconfigs in this PR.

**Comment by boegel** [comment]
Thanks for the review @wpoely86 and @ehiggs!

**Pull request merged**